dulwich/index.py

```diff
@@ -516,7 +516,12 @@ def blob_from_path_and_stat(fs_path, st):
         with open(fs_path, 'rb') as f:
             blob.data = f.read()
     else:
-        blob.data = os.readlink(fs_path)
+        if sys.platform == 'win32':
+            fs_encoding = sys.getfilesystemencoding()
+            blob.data = (os.readlink(fs_path.decode(fs_encoding))
+                         .encode(fs_encoding))
```

Comment:
On Mon, May 25, 2015 at 10:26:58AM -0700, Gary van der Merwe wrote:
> > @@ -516,7 +516,12 @@ def blob_from_path_and_stat(fs_path, st):
> >          with open(fs_path, 'rb') as f:
> >              blob.data = f.read()
> >      else:
> > -        blob.data = os.readlink(fs_path)
> > +        if sys.platform == 'win32':
> > +            fs_encoding = sys.getfilesystemencoding()
> > +            blob.data = (os.readlink(fs_path.decode(fs_encoding))
> > +                         .encode(fs_encoding))
> 
> Maybe I should just work at fixing #203. What are you time lines for the next release?

I'd like to do a new release as soon as we're comfortable enough putting the
Python3 support out there - so probably after the porcelain has been fixed,
and possibly dulwich.web.

Cheers,

Jelmer